### PR TITLE
Create test projects

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Rewrote the readme.
Created test projects in accordance with Arcade naming convention - https://github.com/dotnet/arcade/blob/master/Documentation/ArcadeSdk.md#project-name-conventions.
Dedicate Demo project as canonical example for users.
Some code style updates.

/cc @jkoritzinsky @elinor-fung

### CODE REVIEW

File: DllImportGenerator/readme.md [hunk 1]
```diff
@@ -1,24 +1,33 @@
 # DllImport Generator
 
-See [P/Invoke source generator proposal](https://github.com/dotnet/runtime/blob/master/docs/design/features/source-generator-pinvokes.md) for design and goals.
+Work on this project can be tracked and discussed via the [official issue](https://github.com/dotnet/runtime/issues/43060) in `dotnet/runtime`. The [P/Invoke source generator proposal](https://github.com/dotnet/runtime/blob/master/docs/design/features/source-generator-pinvokes.md) contains additional details.
 
-## Additional work
+## Example
 
-Below are additional work items that are not presently captured in this repository.
+The [Demo project](./DllImportGenerator/Demo) project is designed to be immediately consumable by everyone. It demonstrates a simple use case where the marshalling code is generated and a native function call with only [blittable types](https://docs.microsoft.com/dotnet/framework/interop/blittable-and-non-blittable-types) is made. A managed assembly with [native exports](./DllImportGenerator/TestAssets/NativeExports) is used in the P/Invoke scenario.
```
Comment (elinor-fung) [inline DllImportGenerator/readme.md:7]:
```suggestion
The [Demo project](./DllImportGenerator/Demo) is designed to be immediately consumable by everyone. It demonstrates a simple use case where the marshalling code is generated and a native function call with only [blittable types](https://docs.microsoft.com/dotnet/framework/interop/blittable-and-non-blittable-types) is made. A managed assembly with [native exports](./DllImportGenerator/TestAssets/NativeExports) is used in the P/Invoke scenario.
```

File: DllImportGenerator/readme.md [hunk 2]
```diff
@@ -1,24 +1,33 @@
...
 
-### Required
+### Recommended scenarios:
 
+* Step into the `Demo` application and observe the generated code for the `Sum` functions.
+* Find the implementation of the `sumrefi` function and set a breakpoint. Run the debugger and explore the stack.
+* Add a new export in the `NativeExports` project and update the `Demo` application to call the new export.
+* Consider the differences and try the above scenarios when building in `Debug` or `Release`.
```
Comment (elinor-fung) [inline DllImportGenerator/readme.md:14]:
```suggestion
* Try the above scenarios when building in `Debug` or `Release`. Consider the differences.
```

File: DllImportGenerator/readme.md [hunk 3]
```diff
@@ -1,24 +1,33 @@
...
+- [Struct Marshalling](./designs/StructMarshalling.md)
 
-### Optional
+## Workflow
 
-* A tool to compare the resulting IL from the generated source to that generated by the built-in IL Marshaller system. This would help with validation of what is being generated.
+On Windows, loading the [solution](./DllImportGenerator.sln) in [Visual Studio](https://visualstudio.microsoft.com/) 2019 or later will enable the edit, build, debug inner dev loop. All features of Visual Studio are expected to operate correctly (e.g. Debugger, Test runner, Profiler).
 
-## Designs
+On non-Windows platforms, all features of the `dotnet` command line tool are supported for the respective project types (e.g. `build`, `run`, `test`).
```
Comment (elinor-fung) [inline DllImportGenerator/readme.md:25]: Link to the `dotnet` CLI docs?

File: DllImportGenerator/Demo/Program.cs [hunk 4]
```diff
@@ -31,13 +31,6 @@ static void Main(string[] args)
             c = b;
             NativeExportsNE.Sum(a, ref c);
             Console.WriteLine($"{a} + {b} = {c}");
-
-            SafeHandleTests tests = new SafeHandleTests();
```
Comment (elinor-fung) [inline DllImportGenerator/Demo/Program.cs:35]: Should the project stop referencing `xunit` now?

File: DllImportGenerator/readme.md [hunk 5]
```diff
@@ -1,24 +1,33 @@
...
 
-- [Code generation pipeline](./designs/Pipeline.md)
-- [Struct Marshalling](./designs/StructMarshalling.md)
+A consistent cross-platform inner dev loop is also supported in [Visual Studio Code](https://code.visualstudio.com/) when appropriate .NET extensions are loaded.
+
+Most of the above options have [official tutorials](https://docs.microsoft.com/dotnet/core/tutorials/). It is an aim of this project to follow canonical workflows that are intuitive to all developers.
+
+### Testing assets
+
+This project has no explicit native build system and should remain that way. The [`DNNE`](https://github.com/AaronRobinsonMSFT/DNNE/) project is be leveraged to create native exports that can be called from the P/Invokes during testing.
```
Comment (elinor-fung) [inline DllImportGenerator/readme.md:33]: `is be leveraged`!

File: DllImportGenerator/readme.md (hunk 3 above)
Comment (elinor-fung) [reply DllImportGenerator/readme.md:25]: Also maybe remove `On non-Windows platforms` - makes it seem like this is not supported on Windows.

File: DllImportGenerator/readme.md [hunk 6]
```diff
@@ -1,24 +1,33 @@
...
 
-* A tool to compare the resulting IL from the generated source to that generated by the built-in IL Marshaller system. This would help with validation of what is being generated.
+On Windows, loading the [solution](./DllImportGenerator.sln) in [Visual Studio](https://visualstudio.microsoft.com/) 2019 or later will enable the edit, build, debug inner dev loop. All features of Visual Studio are expected to operate correctly (e.g. Debugger, Test runner, Profiler).
 
-## Designs
+On non-Windows platforms, all features of the `dotnet` command line tool are supported for the respective project types (e.g. `build`, `run`, `test`).
 
-- [Code generation pipeline](./designs/Pipeline.md)
-- [Struct Marshalling](./designs/StructMarshalling.md)
+A consistent cross-platform inner dev loop is also supported in [Visual Studio Code](https://code.visualstudio.com/) when appropriate .NET extensions are loaded.
```
Comment (elinor-fung) [inline DllImportGenerator/readme.md:27]: Should we lead with the consistent cross-platform workflow (instead of starting with the Windows-specific one)?

**Review by elinor-fung — APPROVED** [review]

File: DllImportGenerator/Demo/Program.cs (hunk 4 above)
Comment (AaronRobinsonMSFT) [reply DllImportGenerator/Demo/Program.cs:35]: Yes.

**Pull request merged**